### PR TITLE
Fixed Name Tag Selection/Deselection

### PR DESCRIPTION
**Pull request by gracebarsny**
Desired functionality: when a DogWorldSlot is tapped, its NameTag appears. If the NameTag is tapped, the DogProfile is displayed. A DogWorldSlot is deselected and hides its NameTag if another DogWorldSlot is tapped, or the user otherwise taps outside the DogWorldSlot.

**Prefabs**
- Dog: removed event trigger
- HomeUI: added UI button on content object for deselection of dog world slots

**Scenes**
- DogWorld_Dev: updated to test latest functionality

**Scripts**
- DogWorld: added static ref to selected dog, made dogSlots class variable into local variable, handles deselection of a dog world slot
- DogWorldSlot: removed nameTag.Hide() call (moved to NameTag class)
- NameTag: now hides itself after it's initialized
- DogAI: limited wander radius again

### CODE REVIEW

File: Assets/Scripts/UI/PickupPup/DogWorld/DogWorld.cs [hunk 1]
```diff
@@ -10,22 +10,20 @@ public class DogWorld : MonoBehaviourExtended
 {	
     [SerializeField]
     PPScene room;
+    [SerializeField]
+    UIButton deselectArea;
 
-    DogWorldSlot[] dogsSlots;
+    static DogWorldSlot selectedDogSlot = null;
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/DogWorld/DogWorld.cs]: Is there a reason this needs to be static?

File: Assets/Scripts/UI/PickupPup/DogWorld/NameTag.cs [hunk 2]
```diff
@@ -3,6 +3,7 @@
  * Description: Controls a dog's name tag above its dog world slot.
  */
 
+using UnityEngine;
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/DogWorld/NameTag.cs]: Why did you add this namespace?

Comment (gracebarsny) [inline Assets/Scripts/UI/PickupPup/DogWorld/NameTag.cs]: I think it was to use Debug.Log more easily. I'll remove it.

File: Assets/Scripts/UI/PickupPup/DogWorld/DogWorld.cs (hunk 1 above)
Comment (gracebarsny) [inline Assets/Scripts/UI/PickupPup/DogWorld/DogWorld.cs]: I added it so we could enforce that there would only be one selected dog. I didn't know if there would ever be a case where we would end up with two instances without realizing it that would point to different things, so I avoided that scenario. Personally not attached to it though, if you want it removed.

**Comment by gracebarsny** [comment]
@imann24 I found a bug in revising this PR. Mind giving it another review?

**Comment by imann24** [comment]
@gracebarsny Sure thing. Will do as soon as I submit a PR for the dogs wandering toggle cheat
